src/NHibernate.Test/NHSpecificTest/DtcFailures/DtcFailuresFixture.cs

```diff
@@ -509,6 +509,48 @@ public void CanUseSessionOutsideOfScopeAfterScope([Values(false, true)] bool exp
 			}
 		}
 
+		[Test(Description = "Do not fail, but warn in case a delayed after scope disposal commit is made.")]
+		public void DelayedTransactionCompletion([Values(false, true)] bool explicitFlush)
+		{
+			for (var i = 1; i <= 10; i++)
+			{
+				// Isolation level must be read committed on the control session: reading twice while expecting some data insert
+				// in between due to a late commit. Repeatable read would block and read uncommitted would see the uncommitted data.
+				using (var controlSession = OpenSession())
+				using (controlSession.BeginTransaction(System.Data.IsolationLevel.ReadCommitted))
+				{
+					// We want to have the control session as ready to query as possible, thus beginning its
+					// transaction early for acquiring the connection, even if we will not use it before 
+					// below scope completion.
+
+					using (var tx = new TransactionScope())
+					{
+						using (var s = OpenSession())
+						{
+							s.Save(new Person { CreatedAt = DateTime.Today });
+
+							ForceEscalationToDistributedTx.Escalate();
+
+							if (explicitFlush)
+								s.Flush();
+						}
+						tx.Complete();
+					}
+
+					var count = controlSession.Query<Person>().Count();
+					if (count != i)
+					{
+						Thread.Sleep(100);
+						var countSecondTry = controlSession.Query<Person>().Count();
+						Assert.Warn($"Unexpected entity count: {count} instead of {i}. " +
+							"This may mean current data provider has a delayed commit, occurring after scope disposal. " +
+							$"After waiting, count is now {countSecondTry}. ");
```

Comment:
Does issue a warning with Oracle, and `explicitFlush` `true`, on my laptop. Second try count is good... Since there are not much we can do against such an issue, I let it as just a warning instead of failing.